R/status_checks.R

```diff
@@ -0,0 +1,287 @@
+#' @importFrom tidyr separate_
+#' @importFrom DBI dbGetQuery dbExecute
+#' @importFrom RSQLite dbGetQuery dbWriteTable
+#' @export
+ed_db_check_status <- function(path=NULL) {
+  edb <- eidith_db(path)
+  dbstatus <- ""
+  if(!(all(c(db_tables, db2_tables) %in% db_list_tables(edb$con)))) {
+    #find out which tables are missing and then ask the user if they wish to download them?
+    missing_p1_tables <- db_tables[which(db_tables %in% db_list_tables(edb$con) == FALSE)]
+    missing_p2_tables <- db2_tables[which(db2_tables %in% db_list_tables(edb$con) == FALSE)]
+    dl_p1_tables <- names(purrr::keep(p1_table_names, function(x) x %in% missing_p1_tables))
+    dl_p2_tables <- names(purrr::keep(p2_table_names, function(x) x %in% missing_p2_tables))
+
+    if(interactive()){
+    dl_opt <- menu(c("Yes", "No"), title = "Local EIDITH database is missing tables.\nWould you like to download missing tables?")
+
+    if(dl_opt == 1) ed_db_download(dl_p1_tables, dl_p2_tables)
+    if(dl_opt == 2) dbstatus <- list(status_msg1 ="Local EIDITH database is available, but missing tables.\nRun ed_db_check_status() to update")
+    }else{
+      dbstatus <- list(status_msg1 ="Local EIDITH database is available, but missing tables.\nRun ed_db_check_status() to update")
+    }
+  }else if(!all(sapply(c(db_tables[-7], metadata_tables), function(x) ed_db_field_check(x, NULL)))){
+    #find out which tables have errors
+    error_p1_tables <- sapply(db_tables[-7], function(x) ed_db_field_check(x, NULL))
+    error_p2_tables <- sapply(metadata_tables, function(x) ed_db_field_check(x, NULL))
+    dl_p1_tables <- names(purrr::keep(p1_table_names, function(x) x %in% error_p1_tables))
+    dl_p2_tables <- names(purrr::keep(p2_table_names, function(x) x %in% error_p2_tables))
+
+    if(interactive()){
+    dl_opt <- menu(c("Yes", "No"), title = "Local EIDITH database has tables with corrupt or empty fields.\nWould you like to re-download these tables to correct errors?")
+
+    if(dl_opt == 1) ed_db_download(dl_p1_tables, dl_p2_tables)
+    if(dl_opt == 2) dbstatus <- list(status_msg2 ="Local EIDITH database fields are empty or corrupt. Use ed_db_download() to attempt a clean install.")
+    }else{
+      dbstatus <- list(status_msg ="Local EIDITH database fields are empty or corrupt. Use ed_db_download() to attempt a clean install.")
+    }
+  }else{
+
+    #message(ed_create_banner(path))
+    #class(dbstatus) <- c("dbstatus", class(dbstatus))
+    dbstatus <- "Local EIDITH database contains all tables with all expected fields!"
+  }
+  cat(green(dbstatus))
+  cat_line("")
+}
+
+
+#ON STARTUP CHECK FOR PRESENCE IN STATUS
+ed_db_presence <- function(){
+  edb <- eidith_db()
+  status <- "status" %in% dbListTables(edb$con)
+  if(status == FALSE){
+    cat(cli::rule(crayon::bold("Welcome to the EIDITH R Package!")))
+    cat(red("There is no local EIDITH database, please follow the prompts or use ed_db_download() to download EIDITH data.\n"))

```

Comment:
There was no line break between the welcome message and the no local database message in my terminal.